ghcide/src/Development/IDE/Main.hs

```diff
@@ -249,8 +265,35 @@ defaultMain Arguments{..} = do
                 measureMemory logger [keys] consoleObserver valuesRef
 
             unless (null failed) (exitWith $ ExitFailure (length failed))
+        Index dir -> do
+          dbLoc <- getHieDbLoc dir
+          runWithDb dbLoc $ \hiedb hieChan -> do
+            vfs <- makeVFSHandle
+            sessionLoader <- loadSessionWithOptions argsSessionLoadingOptions dir
+            let options = (argsIdeOptions argsDefaultHlsConfig sessionLoader)
+                        { optCheckParents = pure NeverCheck
+                        , optCheckProject = pure False
+                        }
+            ide <- initialise argsDefaultHlsConfig rules Nothing logger debouncer options vfs hiedb hieChan
+            registerIdeConfiguration (shakeExtras ide) $ IdeConfiguration mempty (hashed Nothing)
+            results <- runAction "Index" ide $ do
+                allKnownTargets <- toKnownFiles <$> useNoFile_ GetKnownTargets
+                liftIO $ hPutStrLn stderr $ "Found " <> show(length allKnownTargets) <> " targets"
+                uses GetModIfaceFromDiskAndIndex $ toList allKnownTargets
```

Comment:
`GetModIfaceFromDiskAndIndex` is async, so all the files will be queued up for indexing, but the rule might return before they are actually indexed. 

We need to wait for `HieDbWriter.indexPending` to be empty to ensure that all files are indexed.

Reply:
Thanks, I missed that